frontend/ui/network/wpa_supplicant.lua

```diff
@@ -69,18 +69,28 @@ function WpaSupplicant:authenticateNetwork(network)
     if err then return false, err end
 
     local re = wcli:setNetwork(nw_id, "ssid", string.format("\"%s\"", network.ssid))
-    if re == 'FAIL' then
+    if re == "FAIL" then
         wcli:removeNetwork(nw_id)
         return false, _("An error occurred while selecting network.")
     end
-    if not network.psk then
-        network.psk = calculatePsk(network.ssid, network.password)
-        self:saveNetwork(network)
-    end
-    re = wcli:setNetwork(nw_id, "psk", network.psk)
-    if re == 'FAIL' then
-        wcli:removeNetwork(nw_id)
-        return false, _("An error occurred while setting password.")
+    -- if password is empty it’s an open AP
+    if network.password and (network.password == nil or string.len(network.password) == 0) then -- Open AP
```

Comment:
`string.len(network.password)` => `#network.password`

(Didn't luacheck complain about that?)

Reply:
Also, `if network.password and network.password == nil` doesn't make any sense?

Reply:
TL;DR: `if network.password and #network.password == 0 then`

Reply:
No luacheck didn’t complain about that